site/src/components/page-faq/contact-modal.tsx

```diff
@@ -0,0 +1,136 @@
+import { Component, h, Prop, State, Watch } from "@stencil/core";
+
+import { Fragment } from "../../util";
+
+import { submitToHelpScout }  from "./helpscout-webhook";
+
+/**
+ * Contact Modal is parent to the contact form
+ * Contains logic to submit form information to helpscout via webhook
+ * Also has logic to show the message submit state and change content based on submit state
+ */
+
+const SUBMIT_SUCCESS = "SUCCESS";
+const SUBMIT_FAILURE = "FAILURE";
+
+const NOT_SUBMITTED = "NOT_SUBMITTED";
+const SUBMITTING = "SUBMITTING";
+const SUBMITTED = "SUBMITTED";
+
+@Component( {
+    tag: "contact-modal",
+    styleUrl: "contact-modal.scss",
+    shadow: false,
+ } )
+export class ContactModal {
+
+    @Prop({
+        mutable: true,
+        reflect: true,
+    })
+    public isOpen: boolean = false;
+
+    @Prop()
+    public onClose?: () => void;
+
+    @State()
+    private submittedState = NOT_SUBMITTED;
+
+    @State()
+    private submitResponseState = "";
+
+    @Watch("isOpen")
```

Comment:
Props and State changes always trigger a re-render so this shouldn't be needed

Reply:
great comment, thank you!